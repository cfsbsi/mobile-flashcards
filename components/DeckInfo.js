import React, {Component} from 'react';
import {View} from 'react-native'
import styled from "styled-components/native/index";
import {cardQuantitiesLiteral} from '../utils/helpers'

class DeckInfo extends Component {

    render() {

        const {title, cardsQuantity} = this.props.navigation.state.params;

        return (
            <ViewStyled>
                <View>
                    <Title>{title}</Title>
                    <SubTitle>{cardQuantitiesLiteral(cardsQuantity)}</SubTitle>
                </View>
                <View>
                    <TouchableOpacityStyledWhite>
                        <TextButtonBlack onPress={() => this.props.navigation.navigate('CardCreation', {title})}>Add
                            Card</TextButtonBlack>
                    </TouchableOpacityStyledWhite>

                    <TouchableOpacityStyledBlack>
                        <TextButtonWhite>Start Quiz</TextButtonWhite>
                    </TouchableOpacityStyledBlack>
                </View>
            </ViewStyled>
        )
    }
}

const ViewStyled = styled.View`
    align-items: center;
    flex: 1;
    justify-content: space-around;
`


const Title = styled.Text`
    font-size: 30;
    font-weight: bold;
    text-align: center;
`

const SubTitle = styled.Text`
    font-size: 15;
    text-align: center;
    font-weight: bold;
    color: gray;
    margin-top: 10;
`


const TouchableOpacityStyledBlack = styled.TouchableOpacity`
    background-color: black;
    border-radius: 7;
    margin-top: 15;
    width: 200;
    padding: 10px 15px;
`

const TouchableOpacityStyledWhite = styled.TouchableOpacity`
    background-color: white;
    border-radius: 7;
    border-width: 2px;
    border-color: black;
    padding: 10px 15px;
`

const TextButtonWhite = styled.Text`
    font-size: 25;
    text-align: center;
    color: white;
`

const TextButtonBlack = styled.Text`
    font-size: 25;
    text-align: center;
    color: black;
`

export default DeckInfo
