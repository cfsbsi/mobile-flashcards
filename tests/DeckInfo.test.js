import React from 'react';
import renderer from 'react-test-renderer';
import DeckInfo from "../components/DeckInfo";

it('renders correctly', () => {
    const state = {state: {params: {title: 'title', subtitle: 'subtitle'}}};
    const tree = renderer.create(<DeckInfo navigation={state}/>).toJSON();
    expect(tree).toMatchSnapshot();
});