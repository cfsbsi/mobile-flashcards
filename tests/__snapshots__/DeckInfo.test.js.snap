// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-around",
      },
      undefined,
    ]
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      title
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "gray",
            "fontSize": 15,
            "fontWeight": "bold",
            "marginTop": 10,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      1 card
    </Text>
  </View>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 7,
          "borderWidth": 2,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 25,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Add Card
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "borderRadius": 7,
          "marginTop": 15,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
          "width": 200,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 25,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Start Quiz
      </Text>
    </View>
  </View>
</View>
`;
